app/src/test/java/waiter/CommunicatorTest.java

```diff
@@ -24,14 +28,31 @@ void setUp() {
 
     @Test
     void shouldAllowSequentialConnections() throws IOException {
+        int numberOfThreadsToGenerate = 1;
         String[] clientRequests = {"curl foo1", "curl foo2", "curl foo3", "curl foo4"};
         ReactorMock reactor = new ReactorMock(clientRequests);
 
         new Communicator(
+                new ThreadGeneratorMock(numberOfThreadsToGenerate),
                 new ListenerMock(new String[]{"foo", clientHasDisconnected}),
                 new MessengerMock()
         ).communicate(reactor);
 
         assertEquals(clientRequests.length, reactor.numberOfAcceptedClients);
     }
+
+    @Property
+    void shouldAllowMultipleConnections(@ForAll @IntRange(min = 1, max = 10) Integer numberOfThreadsToGenerate) throws IOException {
+        String[] clientRequests = {"curl foo1", "curl foo2", "curl foo3", "curl foo4"};
+        ReactorMock reactor = new ReactorMock(clientRequests);
+        ThreadGeneratorMock threadGeneratorMock = new ThreadGeneratorMock(numberOfThreadsToGenerate);
+
+        new Communicator(
+                threadGeneratorMock,
+                new ListenerMock(new String[]{"foo", clientHasDisconnected}),
+                new MessengerMock()
+        ).communicate(reactor);
+
+        assertEquals(clientRequests.length * numberOfThreadsToGenerate, reactor.numberOfAcceptedClients * threadGeneratorMock.numberOfThreadsGenerated);
```

Comment:
Having trouble understanding why these should equal as I don't know what each represents. Could you perhaps place a variable name on them e.g.

someDescriptiveVariableName = clientRequests.length * numberOfThreadsToGenerate
anotherDescriptiveVariableName = reactor.numberOfAcceptedClients * threadGeneratorMock.numberOfThreadsGenerated
assertEquals(someDescriptiveVariableName, anotherDescriptiveVariableName)

Reply:
Addressed in [this commit](https://github.com/jonmanahan/waiter/pull/12/commits/beebfdc3e2c9d8f1314123c5ec4d473e71a203b0), Thank you :D